neuralmonkey/util/word2vec.py

```diff
@@ -33,7 +45,10 @@ def __init__(self, path: str, encoding: str = "utf-8") -> None:
             embedding_vectors.append(np.zeros(emb_size))
             embedding_vectors.append(np.zeros(emb_size))
             # Add placeholder for embedding of the unknown symbol
-            embedding_vectors.append(None)
+            if allow_missing_unk:
+                embedding_vectors.append(np.zeros(emb_size))
```

Comment:
Skutecne staci zeros? Co kdyz budu chtit initializovat embeddingy, ktere budu dale trenovat. Pak tam muzu chtit vektor z nejakeho rozdeleni, ne?

Reply:
V obecnosti samozřejmě ne. Ale imho to pokrývá dva hlavní usacasy:

1. mám slovní embedingy, které by měly unk obsahovat
2. mám embedingy sub-wordů a unk by nikdy neměl objevit, tak je více méně jedno, co tam bude za vektor

Reply:
Ostatně stejně by se dalo argumentovat pro ty ostatní symboly, pro které se nastavujou nuly.

Reply:
To dalo. Je otazka, jestli by si user spis chybejici symboly nemel doplnit rucne do toho W2V file, resp. to tak stejne bude muset udelat, pokud to tak bude chtit.

Kdyz uz, tak bych ten orig.`None` placeholder nahradil np.zeros placeholderem a zahodil flag allow_missing_unk.

Btw, vsechny slusne metody (jejich implementace) na delani embeddingu unk podpodporuji minimalne jako optional, ne?

Reply:
Tak use udělal nuly default. Stejně mi moc nebylo jasný, proč se ten unk checkuje.

Ad všechny metody: když má něco za úkol dělat embedingy, tak asi jo. Když si bereš embedingy s lecčeho, tak být nemusí.